File: src/GymManagement.Domain/Entities/BodyPart.cs
namespace GymManagement.Domain.Entities
{
    public class BodyPart : BaseEntity
    {
        public string Name { get; set; } = string.Empty;
        public string? Description { get; set; }
        public string? ImageUrl { get; set; }

        // 3D Model Camera Position (stored as JSON)
        public string? CameraPositionJson { get; set; }

        public ICollection<BodyPartMuscle> BodyPartMuscles { get; set; } = new List<BodyPartMuscle>();
        public ICollection<Exercise> Exercises { get; set; } = new List<Exercise>();
    }
}

File: gym_client/src/types/bodyPart.ts
export interface BodyPartMuscle {
  id: number
  bodyPartId: number
  name: string
  description?: string | null
  imageUrl?: string | null
}

export interface BodyPart {
  id: number
  name: string
  description?: string | null
  imageUrl?: string | null
  exerciseCount: number
  bodyPartMuscles: BodyPartMuscle[]
  cameraPositionJson?: string | null
}

export interface CreateBodyPartDto {
  name: string
  description?: string | null
  imageUrl?: string | null
}

export interface UpdateBodyPartDto {
  name: string
  description?: string | null
  imageUrl?: string | null
}

export interface CreateBodyPartMuscleDto {
  bodyPartId: number
  name: string
  description?: string | null
  imageUrl?: string | null
}

export interface UpdateBodyPartMuscleDto {
  name: string
  description?: string | null
  imageUrl?: string | null
}
